keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+        //
+        // +optional
+        PodNetworkName string `json:"podNetworkName,omitempty"`
+
+        // PodNetworkAttachmentName is name of PodNetwork to attach
+        // Only one of: [PodNetworkName, PodNetworkAttachmentName] can be set
+        //
+        // +optional
+        PodNetworkAttachmentName string `json:"podNetworkAttachmentName,omitempty"`
+
+        // InterfaceName is the network interface name inside the Pod for this attachment.
+        // This field functionality is dependent on the implementation and its support
+        // for it.
+        // Examples: eth1 or net1
+        //
+        // +optional
+        InterfaceName string `json:"interfaceName,omitempty"`
+
+        // IsDefaultGW4 is a flag indicating this PodNetwork will hold the IPv4 Default
+        // Gateway inside the Pod. Only one Network can have this flag set to True.
+        // This field functionality is dependent on the implementation and its support
+        // for it.
+        //
+        // +optional
+        IsDefaultGW4 bool `json:"isDefaultGW4,omitempty"`
+
+        // IsDefaultGW6 is a flag indicating this PodNetwork will hold the IPv6 Default
+        // Gateway inside the Pod. Only one Network can have this flag set to True.
+        // This field functionality is dependent on the implementation and its support
+        // for it.
+        //
+        // +optional
+        IsDefaultGW6 bool `json:"isDefaultGW6,omitempty"`
+}
+```
+
+#### Static validations
+We will perform static validation of the provided spec in the API Server,
+alongside the current Pod spec checks. These errors will be provided to the user
+immediately when they try to create Pod.
+This will include:
+* Ensure a single Pod references a given PodNetwork only 1 time
+* IsDefaultGW4 and IsDefaultGW6 uniqueness for “true” value across multiple
+“Network” objects
+* InterfaceName uniqueness across multiple “Network” objects
+* InterfaceName naming constraints for Linux and Windows
+* Ensure Network objects are not specified when hostNetwork field is set
+
+The 1 PodNetwork per Pod restriction is coming from our current lack of details
+for future Multi-Networking requirements (e.g. Service support), and can be
+changed in future when we will discuss them.
+
+#### Active validations
+Beside the above (static validations), we will perform additional active
+validations, inside the  scheduler, that require queries for other objects
+(e.g. PodNetwork). These errors will be presented as Pod Events, and the Pod
+will be kept in Pending state until the issues are resolved. We will do the
+following validation:
+* Referenced PodNetwork or PodNetworkAttachment is present
+* Referenced PodNetwork or PodNetworkAttachment is Ready
+* Referenced PodNetworkAttachment, that is referencing a PodNetwork, follows the
+rule of: single Pod references a given PodNetwork only 1 time
+
+#### Auto-population
+When networks field is not set, and hostNetwork is not set, we will
```

Comment:
what happens with pods with hostNetwork: true? to which network do they belong?

Reply:
None, MN does not apply to HN pods.

Reply:
so how does this work then, if I have 3 pods, podA in networkA pod B in networkB and podC in host network, which Pods are able to communicate with podC?

Reply:
Because Today there is only 1 network, root ns and Pod ns is treated as one. And we need to still assume that, and we call this a "default" Network, pods connected to that Network has to be backward compatible. Any other networks behavior is up the implementation. For some, isolation is a factor for some it is not, we cannot enforce that.